components/bookshelf/Book.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from "react"
+import Image from "next/image"
+import {
+  BookCard,
+  BookContent,
+  BookImage,
+  HeaderBar,
+  InfoBox,
+  Author,
+  Publisher,
+  Title,
+  Synopsis,
+  SideBar,
+  ExtendedSideBar,
+} from "../styled/bookshelf"
+import { BookInterface } from "../../api/types"
+
+interface BookProps {
+  book: BookInterface
+}
+
+const Book = ({ book }: BookProps) => {
+  const [isAdding, setIsAdding] = useState<boolean>(false)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
```

Comment:
Why do you have 2 loadng states?

Reply:
yep. that's not right 😅 here's what I did.

made an 'enum'
![image](https://user-images.githubusercontent.com/91857581/188650769-81840928-95f6-4429-be0f-d70ee5d1888b.png)

imported it as a state in the parent component and passed the props to it
![image](https://user-images.githubusercontent.com/91857581/188650946-83c2ee3b-8743-434a-8889-8dde31171886.png)

1. I needed a way to tell the Book component to show a specific set of buttons when its rendered in the Reading List and Finished Books components.
2. I needed to be able to change the state from inside the Book component to set and de-set the 'loading' animation when making the api call.